rna/input_output.py

```diff
@@ -0,0 +1,250 @@
+        df = df_rv.loc[:, (df_rv.columns.values[:-1])]
+        rv = df_rv[['replicate_value']]
+        df.fillna(0, inplace=True)
+        if binarize:
+            df = 1 * (df > 150)
+        return df, rv
+
+    else:
+        # TODO: What type of data is to be expected?
+        # TODO: Add replicate numbers manually
+        # then 'replicate_values' not included, so
+        # assume that all samples have 4 replicates
+        df = pd.read_excel(filename, delimiter=';', index_col=0)
+        df.fillna(0, inplace=True)
+        if binarize:
+            df = 1 * (df > 150)
+
+        try:
+            rv = np.zeros((df.shape[0], 1))
+            replicates = [i for i in range(number_of_replicates)] * int(df.shape[0]/number_of_replicates)
+            rv[:, 0] = replicates
+        except ValueError:
+            rv[:, 0] = [i for i in range(number_of_replicates)] * int(df.shape[0] / number_of_replicates) + \
+                       [i for i in range(number_of_replicates)][0:df.shape[0] - len(replicates)]
+        return df, rv
+
+
+def get_data_per_cell_type(filename='Datasets/Dataset_NFI_rv.xlsx', single_cell_types=None,
+                           ground_truth_known=True, binarize=True, number_of_replicates=None):
+    """
+    Returns data per specified cell types.
+
+    Note that the samples are saved in a separate numpy array
+    based on their replicate indices. As the size differs per
+    sample, the array is filled up with zeros to get the correct
+    dimension.
+
+    :param filename: name of file to read in, must include "_rv"
+    :param single_cell_types: iterable of strings of all single cell types that exist
+    :param ground_truth_known: does this data file have labels for the real classes?
+    :param binarize: whether to binarize raw measurement values
+    :return: (N_single_cell_experimental_samples x N_measurements per sample x
+        N_markers array of measurements,
+                N_single_cell_experimental_samples array of int labels of which
+                    cell type was measured,
+                N_cell types,
+                N_markers (=N_features),
+                dict: cell type name -> cell type index,
+                dict: cell type index -> cell type name,
+                dict: cell type index -> N_measurements for cell type
+
+    """
+
+    df, rv = read_df(filename, binarize, number_of_replicates)
+
+    label_encoder = LabelEncoder()
+    if single_cell_types:
+        single_cell_types = list(set(single_cell_types))
+        single_cell_types.append('Skin.penile')
+        label_encoder.fit(single_cell_types)
+    else:
+        if not ground_truth_known:
+            raise ValueError('if no cell types are provided, ground truth should be known')
+        # if not provided, learn the cell types from the data
+        all_celltypes = np.array(df.index)
+        for celltype in all_celltypes:
+            # TODO: How does this work if single_cell_types is None?
```

Comment:
de mooiere oplossing voor de penile is overigens om in constants een lijst van 'environmental_cell_types' oid te definieren, die voor nu alleen bestaat uit Skin.penile. Dan kan je die hele lijst als uitzondering meenemen, ipv over 'penile' uitzonderingen. Feitelijk zijn dat celtypes waarvan je a priori al weet dat het voor dit model altijd kans 1 heeft - een lijst die we dus wellicht nog uit willen breiden.